.github/workflows/docs.yml

```diff
@@ -18,13 +18,64 @@ name: "Documentation check"
 on:
 - pull_request
 
+env:
+  DEPS_BRANCH: bot/stable-deps-update
+
 jobs:
   sphinx:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
     - uses: PennyLaneAI/sphinx-action@master
       with:
         docs-folder: "doc/"
-        pre-build-command: "apt install -y graphviz && pip3 install -r doc/requirements.txt && pip3 install . pip3 install openfermionpyscf && pip3 install aiohttp fsspec h5py"
+        pre-build-command: "apt install -y graphviz && pip3 install -r doc/requirements.txt && pip3 install . && pip3 install openfermionpyscf && pip3 install aiohttp fsspec h5py && pip freeze | grep -v 'file:///' > .github/stable/doc.txt.tmp"
         build-command: "sphinx-build -b html . _build -W --keep-going"
+
+    - name: Prepare local repo
+      run: |
+        git fetch
+        git config user.name "GitHub Actions Bot"
+        git config user.email "<>"
+        if git ls-remote --exit-code origin "refs/heads/${{ env.DEPS_BRANCH }}"; then
+          git checkout "${{ env.DEPS_BRANCH }}"
+        else
+          git checkout master
+          git pull
+          git checkout -b "${{ env.DEPS_BRANCH }}"
+        fi
+        mv -f .github/stable/doc.txt.tmp .github/stable/doc.txt
+
+    - name: Determine if changes have been made
+      id: changed
+      run: |
+        echo "has_changes=$(git status --porcelain | wc -l | awk '{print $1}')" >> $GITHUB_OUTPUT
+
+    - name: Stage changes
+      if: steps.changed.outputs.has_changes != '0'
+      run: |
+        git add .github/stable/doc.txt
+        git commit -m "Update stable docs dependencies"
+        git push -f --set-upstream origin "${{ env.DEPS_BRANCH }}"
+
+    # Create PR to master
+    - name: Create pull request
```

Comment:
# Question

I missed this last time, but what happens if a PR already exists? Nothing happens?